src/backend/tests/it/bitcoin.rs

```diff
@@ -72,7 +72,10 @@ fn test_select_user_utxos_fee_pending_transaction_error() {
         add_request,
     );
 
-    assert!(add_response.is_ok());
+    assert!(
+        add_response.is_ok(),
+        "btc_add_pending_transaction failed: {add_response:?}"
+    );
```

Comment:
This failed and the error message was not helpful, hence the increased verbosity.

The test failed because the toy transaction ID is empty and the validator checked for exactly 32 bytes.

I have relaxed the validation to allow no more than 32 bytes, to limit how many changes are needed in this one PR, however it would be good to make the validation a bit stricter, eventually.  The backend is just a dumb store and the only thing it really needs to care about is being overwhelmed with data.